src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -888,6 +888,8 @@ export class NotebookModel extends Disposable implements INotebookModel {
 	public changeKernel(displayName: string): void {
 		this._selectedKernelDisplayName = displayName;
 		this._currentKernelAlias = this.context?.serverCapabilities?.notebookKernelAlias;
+		this.adstelemetryService.createActionEvent(TelemetryKeys.TelemetryView.Notebook, TelemetryKeys.NbTelemetryAction.ChangeKernel)
```

Comment:
You're still using the localized displayName here. Is there something preventing you from putting this somewhere like line 924 after we get the IKernelSpec for the selected kernel? (or even in the client session instead)